File: src/main/java/ae/teletronics/nlp/language/detection/TikaDetector.java
package ae.teletronics.nlp.language.detection;

import com.neovisionaries.i18n.LanguageCode;
import org.apache.tika.language.LanguageIdentifier;

import java.util.List;

/**
 * Created by trym on 10-02-2016.
 */
public class TikaDetector implements LanguageDetector {

    @Override
    public LanguageCode detect(String text) {
        return LanguageMapper.fromString(new LanguageIdentifier(text).getLanguage());
    }

    @Override
    public List<LanguageProbability> detectAll(String text) {
        throw new UnsupportedOperationException();
    }
}

File: src/main/scala/ae/teletronics/nlp/language/detection/detectors/TikaDetector.scala
package ae.teletronics.nlp.language.detection.detectors

import ae.teletronics.nlp.language.detection.model.Language
import com.neovisionaries.i18n.LanguageCode
import org.apache.tika.language.LanguageIdentifier

/**
  * Created by trym on 18-02-2016.
  */
class TikaDetector extends SLanguageDetector {
  override def minimalConfidence() = 0.5

  override def detect(text: String): java.util.List[Language] = {
    import scala.collection.JavaConversions._
    val languageIdentifier = new LanguageIdentifier(text)
    if (languageIdentifier.isReasonablyCertain()) {
      List(new Language(LanguageCode.getByCodeIgnoreCase(languageIdentifier.getLanguage()), 0.5))
    } else {
      List.empty[Language]
    }
  }
}
